### PR TITLE
[CP-965] Implementing template updating UI

### PR DESCRIPTION
**Pull request by igorbezsmertnyi**
Jira: [CP-965]

**Description**
- Implement update templates update
- Separate modal logic
- Implement state management hook 

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

[CP-965]: https://appnroll.atlassian.net/browse/CP-965?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/templates/components/template-options/template-options.component.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { IconType } from "App/__deprecated__/renderer/components/core/icon/icon-type"
+import { ElementWithTooltipPlace } from "App/__deprecated__/renderer/components/core/tooltip/element-with-tooltip.component"
+import ButtonComponent from "App/__deprecated__/renderer/components/core/button/button.component"
+import { DisplayStyle } from "App/__deprecated__/renderer/components/core/button/button.config"
+import { TemplateOptionsProps } from "App/templates/components/template-options/template-options.interface"
+import { TemplateOptionsTestIds } from "App/templates/components/template-options/template-options-test-ids.enum"
+
+const messages = defineMessages({
+  dropdownTogglerTooltipDescription: {
+    id: "component.dropdownTogglerTooltipDescription",
```
Comment (wojcik92michal) [inline packages/app/src/templates/components/template-options/template-options.component.tsx:21]: Is it the correct id? I couldn't find it in `en-US.json`.

image

File: packages/app/src/templates/components/template-options/template-options.component.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
```
Comment (wojcik92michal) [inline packages/app/src/templates/components/template-options/template-options.component.tsx:3]: (adding it here as it is easier to have a thread)

I cannot add any new template. Tested on MuditaOS 1.2.0 - do you think that is the reason?

https://user-images.githubusercontent.com/41268731/173573155-23eaad02-c81d-4855-83ea-97c3aaa1037b.mov

Comment (igorbezsmertnyi) [reply packages/app/src/templates/components/template-options/template-options.component.tsx:3]: You probably should use the latest daily build of MuditaOS. The official release doesn't contains those changes

File: packages/app/src/templates/services/template.service.ts [hunk 3]
```diff
@@ -107,7 +107,7 @@ export class TemplateService {
       body: TemplatePresenter.mapToPureTemplateBody(template),
     })
 
-    if (isResponseSuccessWithData(response)) {
+    if (response.status === RequestResponseStatus.Ok && !response.error) {
```
[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/templates/services/template.service.ts]: https://github.com/mudita/mudita-center/pull/781/commits/30f248ade0d33de6df559ab3a4434c36f7acfb9c

File: packages/app/src/templates/components/template-options/template-options.component.tsx (hunk 1 above)
Comment (igorbezsmertnyi) [reply packages/app/src/templates/components/template-options/template-options.component.tsx:21]: https://github.com/mudita/mudita-center/pull/781/commits/7498e616c570ec4a60e65142608e52b9011156a1

File: packages/app/src/templates/components/deleting-template-modals/deleting-template-modals.component.test.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    ).not.toBeInTheDocument()
+    expect(
+      queryByTestId(DeletingTemplateModalsTestIds.LoadingModal)
+    ).not.toBeInTheDocument()
+    expect(
+      queryByTestId(DeletingTemplateModalsTestIds.ErrorModal)
+    ).not.toBeInTheDocument()
+  })
+
+  test("displays info pop up with deleted entity count if `creatingInfo` is equal to `true`, `error` is empty and `deletedTemplatesLength` more then `1`", () => {
```
[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/templates/components/deleting-template-modals/deleting-template-modals.component.test.tsx]: https://github.com/mudita/mudita-center/pull/781/commits/48a14234133011c77e9b6c7748337a66d26a4a55

File: packages/app/src/templates/actions/update-template.action.test.ts [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+      expect(mockStore.getActions()).toEqual([
+        updateTemplate.pending(requestId, template),
+        updateTemplate.rejected(testError, requestId, template, errorMock),
+      ])
+
+      expect(updateTemplateRequest).toHaveBeenCalled()
+    })
+
+    test("fire async `createTemplate` returns `rejected` action if error is exist", async () => {
```
[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/templates/actions/update-template.action.test.ts]: https://github.com/mudita/mudita-center/pull/781/commits/48a14234133011c77e9b6c7748337a66d26a4a55

File: packages/app/src/templates/actions/update-template.action.test.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+beforeEach(() => {
+  mockStore.clearActions()
+})
+
+jest.mock("App/templates/requests/update-template.request")
+
+describe("async `updateTemplate`", () => {
+  describe("when `updateTemplateRequest` requests return success", () => {
+    test("fire async `updateTemplate` returns update template", async () => {
```
[the remark this answers is not in the document]
Comment (igorbezsmertnyi) [reply packages/app/src/templates/actions/update-template.action.test.ts]: https://github.com/mudita/mudita-center/pull/781/commits/48a14234133011c77e9b6c7748337a66d26a4a55

**Review by dkarski — APPROVED** [review]

**Pull request merged**